components/gitpod-db/go/cost_center.go

```diff
@@ -122,6 +119,78 @@ func (c *CostCenterManager) GetOrCreateCostCenter(ctx context.Context, attributi
 	return result, nil
 }
 
+// computeDefaultSpendingLimit computes the spending limit for a new Organization.
+// If the first joined member has not already granted credits to another org, we grant them the the free credits allowance.
+func (c *CostCenterManager) getSpendingLimitForNewCostCenter(attributionID AttributionID) int32 {
+	_, orgId := attributionID.Values()
+	orgUUID, err := uuid.Parse(orgId)
+	if err != nil {
+		log.WithError(err).WithField("attributionId", attributionID).Error("Failed to parse orgId.")
+		return c.cfg.ForTeams
+	}
+
+	// fetch the first owner that joined the org
+	var userId string
+	db := c.conn.Raw(`
+		SELECT userid
+		FROM d_b_team_membership
+		WHERE
+			teamId = ? AND
+			creationTime = (
+				SELECT min(creationTime)
+				FROM d_b_team_membership
+				WHERE
+					role='owner' AND
+					deleted = 0 AND
+					teamID = ?
+			)
+	`, orgId, orgId).Scan(&userId)
+	if db.Error != nil {
+		log.WithError(db.Error).WithField("attributionId", attributionID).Error("Failed to get userId for org.")
+		return c.cfg.ForTeams
+	}
+
+	if userId == "" {
+		log.WithField("attributionId", attributionID).Error("Failed to get userId for org.")
+		return c.cfg.ForTeams
+	}
+
+	userUUID, err := uuid.Parse(userId)
+	if err != nil {
+		log.WithError(err).WithField("attributionId", attributionID).Error("Failed to parse userId for org.")
+		return c.cfg.ForTeams
+	}
+
+	// check if the user has already granted free credits to another org
+	type FreeCredit struct {
```

Comment:
Pull this outside of the method, such that it can be accessed in general? Want to include the Last modified as a field as well?

Reply:
it is intended to be only used within that method for now.

Reply:
Don't we need unit tests which check the record exists in the DB?

Reply:
I'm testing it one level of abstraction above (i.e. the cost center).